components/opponent/wikis/stormgate/opponent_display_custom.lua

```diff
@@ -0,0 +1,157 @@
+	opponent = MatchGroupUtil.types.GameOpponent,
+	hideFaction = 'boolean?',
+})
+
+---Display component for an opponent entry appearing in a bracket match.
+---@class StormgateBracketOpponentEntry
+---@operator call(...): StormgateBracketOpponentEntry
+---@field content Html
+---@field root Html
+CustomOpponentDisplay.BracketOpponentEntry = Class.new(
+	---@param self self
+	---@param opponent StormgateStandardOpponent
+	---@param options {forceShortName: boolean}
+	function(self, opponent, options)
+		local showFactionBackground = opponent.type == Opponent.solo or opponent.extradata.hasFactionOrFlag
+
+		self.content = mw.html.create('div'):addClass('brkts-opponent-entry-left')
+			:addClass(showFactionBackground and Faction.bgClass(opponent.players[1].faction) or nil)
+
+		if opponent.type == Opponent.team then
+			self.content:node(OpponentDisplay.BlockTeamContainer({
+				showLink = false,
+				style = 'hybrid',
+				team = opponent.team,
+				template = opponent.template,
+			}))
+		else
+			self.content:node(CustomOpponentDisplay.BlockOpponent({
+				opponent = opponent,
+				overflow = 'ellipsis',
+				playerClass = 'starcraft-bracket-block-player',
+				showLink = false,
+			}))
+		end
+
+		self.root = mw.html.create('div'):addClass('brkts-opponent-entry')
+			:node(self.content)
+	end
+)
+
+CustomOpponentDisplay.BracketOpponentEntry.addScores = OpponentDisplay.BracketOpponentEntry.addScores
+
+---@class StormgateInlineOpponentProps: InlineOpponentProps
+---@field opponent StormgateStandardOpponent
+---@field hideFaction boolean?
+
+---@param props StormgateInlineOpponentProps
+---@return Html|string|nil
+function CustomOpponentDisplay.InlineOpponent(props)
+	DisplayUtil.assertPropTypes(props, CustomOpponentDisplay.propTypes.InlineOpponent)
+	local opponent = props.opponent
+
+	if Opponent.typeIsParty((opponent or {}).type) then
+		return CustomOpponentDisplay.InlinePlayers(props)
+	end
+
+	return OpponentDisplay.InlineOpponent(props)
+end
+
+CustomOpponentDisplay.propTypes.BlockOpponent = TypeUtil.extendStruct(OpponentDisplay.propTypes.BlockOpponent, {
+	opponent = MatchGroupUtil.types.GameOpponent,
+	hideFaction = 'boolean?',
+	playerClass = 'string?',
+})
+
+---@class StormgateBlockOpponentProps: BlockOpponentProps
+---@field opponent StormgateStandardOpponent
+---@field hideFaction boolean?
+
+---@param props StormgateBlockOpponentProps
+---@return Html
+function CustomOpponentDisplay.BlockOpponent(props)
+	DisplayUtil.assertPropTypes(props, CustomOpponentDisplay.propTypes.BlockOpponent)
+	local opponent = props.opponent
+
+	opponent.extradata = opponent.extradata or {}
+	-- Default TBDs to not show links
+	local showLink = Logic.nilOr(props.showLink, not Opponent.isTbd(opponent))
+
+	if opponent.type == Opponent.literal and opponent.extradata.hasFactionOrFlag then
+		return CustomOpponentDisplay.BlockPlayers(Table.merge(props, {showLink = showLink}))
+	elseif Opponent.typeIsParty((opponent or {}).type) then
```

Comment:
Redudant `or {}`, as it will error on the first if conditional if opponent is nil